src/Sulu/Bundle/AdminBundle/Resources/js/containers/Badge/stores/BadgeStore.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+import {observable, action, autorun} from 'mobx';
+import jexl from 'jexl';
+import jsonPointer from 'json-pointer';
+import symfonyRouting from 'fos-jsrouting/router';
+import Router from '../../../services/Router';
+import Requester from '../../../services/Requester';
+
+export default class BadgeStore {
+    router: Router;
+    routeName: string;
+    dataPath: ?string;
+    visibleCondition: ?string;
+    attributesToRequest: Object;
+    routerAttributesToRequest: Object;
+    @observable text: ?string = null;
+    disposer: () => {};
+
+    constructor(
+        router: Router,
+        routeName: string,
+        dataPath: ?string,
+        visibleCondition: ?string,
+        attributesToRequest: Object,
+        routerAttributesToRequest: Object
+    ) {
+        this.router = router;
+        this.routeName = routeName;
+        this.dataPath = dataPath;
+        this.visibleCondition = visibleCondition;
+        this.attributesToRequest = attributesToRequest;
+        this.routerAttributesToRequest = routerAttributesToRequest;
+
+        this.disposer = autorun(() => {
```

Comment:
Initially and on tab change ...

Reply:
but why is it run on tab change this way? 🤔

Reply:
hmm ... I have no idea 😄

Reply:
Currently, the autorun only triggers when changing tabs, not when a new Task is added or deleted. Do you have an idea what else we should listen to to update the badge when new data is available? But we don't have access to the ResourceStore nor the ListStore

Reply:
I think we only have access to the router

Reply:
The bigger problem is, that the badge isn't updated after deleting a task, because deleting is triggered by the `DeleteToolbarAction` on the list view, so the url won't change